lib/AST/UnqualifiedLookup.cpp

```diff
@@ -0,0 +1,1806 @@
+                                          Optional<bool> isCascadingUse);
+
+  void lookForLocalVariablesIn(AbstractFunctionDecl *AFD,
+                               Optional<bool> isCascadingUse);
+  void lookForLocalVariablesIn(ClosureExpr *);
+  void lookForLocalVariablesIn(SourceFile *);
+
+  bool isOutsideBodyOfFunction(const AbstractFunctionDecl *const AFD) const;
+
+  void addGenericParametersHereAndInEnclosingScopes(DeclContext *dc);
+  void addGenericParametersHereAndInEnclosingScopes(GenericParamList *);
+
+  /// Consume generic parameters
+  void addGenericParametersForFunction(AbstractFunctionDecl *AFD);
+
+  static GenericParamList *getGenericParams(const DeclContext *const dc);
+
+  /// For diagnostic purposes, move aside the unavailables, and put
+  /// them back as a last-ditch effort.
+  /// Could be cleaner someday with a richer interface to UnqualifiedLookup.
+  void setAsideUnavailableResults(size_t firstPossiblyUnavailableResult);
+
+  void recordDependencyOnTopLevelName(DeclContext *topLevelContext,
+                                      DeclName name, bool isCascadingUse);
+
+  void addPrivateImports(DeclContext *const dc);
+
+  void addNamesKnownToDebugClient(DeclContext *dc);
+
+  void addUnavailableInnerResults();
+
+  void lookForAModuleWithTheGivenName(DeclContext *const dc);
+
+#pragma mark common helper declarations
+  static NLOptions
+  computeBaseNLOptions(const UnqualifiedLookup::Options options,
+                       const bool isOriginallyTypeLookup);
+
+  static bool resolveIsCascadingUse(const DeclContext *const dc,
+                                    Optional<bool> isCascadingUse,
+                                    bool onlyCareAboutFunctionBody);
+  static bool resolveIsCascadingUse(DCAndUnresolvedIsCascadingUse x,
+                                    bool onlyCareAboutFunctionBody) {
+    return resolveIsCascadingUse(x.whereToLook, x.isCascadingUse,
+                                 onlyCareAboutFunctionBody);
+  }
+
+  void findResultsAndSaveUnavailables(ResultFinderForSelfsConformances &&PTS,
+                                      bool isCascadingUse,
+                                      NLOptions baseNLOptions,
+                                      DeclContext *lookupContextForThisContext);
+
+  void dumpBreadcrumbs() const;
+
+public:
+  bool verifyEqualToLegacy(const LegacyUnqualifiedLookup &&LUL) const;
+};
+} // namespace
+
+#pragma mark UnqualifiedLookupFactory functions
+
+// clang-format off
+UnqualifiedLookupFactory::UnqualifiedLookupFactory(
+                                                   DeclName Name,
+                                                   DeclContext *const DC,
+                                                   LazyResolver *TypeResolver,
+                                                   SourceLoc Loc,
+                                                   Options options,
+                                                   UnqualifiedLookup &lookupToBeCreated)
+:
```

Comment:
This formatting where the member initializations are at column zero makes my cringe ;)

Reply:
Oops! Yes, I wanted one per line, but not at zero. I can move them to col 2, but is there a better way? Clang-format wants to pack >1 per line...